src/main/java/uk/gov/register/indexer/function/AddAllIndexFunction.java

```diff
@@ -7,15 +7,13 @@
 
 import java.util.Set;
 
-public class MetadataIndexFunction extends BaseIndexFunction {
-    public MetadataIndexFunction(String name) {
+public class AddAllIndexFunction extends BaseIndexFunction {
```

Comment:
Perhaps we rename this to something like `LatestByKey`, given that the index function isn't controlling what is and isn't being appended to the register?